src/full/Agda/Compiler/MAlonzo/Compiler.hs

```diff
@@ -195,8 +195,6 @@ data GHCDefinition = GHCDefinition
 
 ghcPreCompile :: GHCFlags -> TCM GHCCompileEnv
 ghcPreCompile flags = do
-  allowUnsolved <- optAllowUnsolved <$> pragmaOptions
-  when allowUnsolved $ genericError $ "Unsolved meta variables are not allowed when compiling."
```

Comment:
@gallais : I remove this check which would stop compilation for --allow-unsolved-metas.  In absence of this flag, we do not even get to compilation, but fail with error "Unsolved metas" before.